test/controllers/sessions_controller_test.rb

```diff
@@ -0,0 +1,45 @@
+require "test_helper"
+
+describe SessionsController do
+  describe "create" do
+    it "should log in an existing user and redirect to homepage" do
+      before_count = Merchant.count
+      existing_merchant = merchants(:lucy)
+
+      perform_login(existing_merchant)
+
+      must_redirect_to homepage_path
+      Merchant.count.must_equal before_count
+    end
+
```

Comment:
What about if the route is hit without the proper information to log in.  Like the user bypasses github and goes to the route.